docs/api/implementation-guide.md

```diff
@@ -0,0 +1,100 @@
+---
+title: NuGet Server Implementation Guide
+description: Guidelines and recommendations to anyone implementing the NuGet Server API in their own package repository
+author: zivkan
+ms.author: zivkan
+ms.date: 07/29/2023
+ms.topic: conceptual
+---
+
+# NuGet Server Implementation Guide
+
+Prior to 2018, NuGet package repositories could copy the package metadata from the `nupkg`'s [`nuspec` file](../reference/nuspec.md), and return the results unmodified in [search](./search-query-service-resource.md) and [package metadata](./registration-base-url-resource.md) requests.
```

Comment:
This is a pretty strong statement. The shapes are quite different and in practice data manipulation occurs even for the fields that overlap between the XML .nuspec and the JSON search response. For example, the tags returned is an array, version normalization occurs (.nuspec is the input, but is not guaranteed to be normalize), the version list in search results is not available in the .nuspec (a nuspec is a version specific thing), etc etc

Perhaps I am missing some intent of the document but it reads a bit oddly to me.

What about an opening that starts a bit broader, excludes the nuance of history, while still alluding to the fact the server data mostly comes from package metadata. How about:

> In some cases, you may want to implement your own NuGet package feed. [Many existing implementations](https://learn.microsoft.com/en-us/nuget/hosting-packages/overview) exist which allow you to host your own feed in a variety way, but the protocol between the official NuGet client software and a package feed is documented allowing you to build your own feed implementation from scratch.
>
> The protocol does evolve over time and this guide is aimed at those that wish to or already have implemented a NuGet package server.
>
> Since the initial release of the NuGet V3 protocol in 2015, NuGet has evolved to provide developers with a richer experience, and this requires package repositories to do additional work in order to provide the additional value their package consumers, beyond simply exacting metadata from hosted packages and returning the metadata in various forms. For example, the [search](./search-query-service-resource.md) and [package metadata](./registration-base-url-resource.md) endpoints contain more than just metadata found in the `nupkg`'s [`nuspec` file](../reference/nuspec.md).
> 
> Note that this guide is focused on the NuGet V3 protocol since the V2 protocol is essentially undocumented and, since 2015, the recommended protocol for NuGet client and server communication is the V3 protocol. For more information read about [protocol versioning](https://learn.microsoft.com/en-us/nuget/api/overview#versioning).

Feel free to scrap my wording but the current intro focuses on one very specific confusion point which I think is too specific for such a broad guide.

Reply:
I've spent so many hours on this document now, I really need to focus on my other work, so I copy-pasted your text. To be perfectly honest, I think this is too long of an introduction. We all know that the more we write, the sooner we lose the attention of people who need to read it, so shorter is better. But I really want to get this merged and move on to other work, so maybe someone can open a PR to improve it after this PR is merged.